src/App.tsx

```diff
@@ -1,26 +1,43 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Todo } from './types/Todo';
+import { todosSlice } from './features/todos';
+import { getTodos } from './api';
+import { useAppSelector } from './app/hooks';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [loading, setLoading] = useState(false);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useDispatch();
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then((data: Todo[]) => dispatch(todosSlice.actions.setTodos(data)))
+      .finally(() => setLoading(false));
```

Comment:
You are not handling any errors that might occur during the fetching of todos. If the `getTodos` function fails for any reason (e.g., the server is down, the client is offline), the promise will reject, and an uncaught promise rejection will occur, which is not a good practice. Consider adding a `.catch` block to handle any errors that might occur during the fetch operation. You could update the state to indicate that an error has occurred and perhaps display a message to the user.